src/transforms/group.js

```diff
@@ -444,6 +445,7 @@ export function find(test) {
   if (typeof test !== "function") throw new Error(`invalid test function: ${test}`);
   return {
     reduceIndex(I, V, {data}) {
+      data = arrayify(data);
```

Comment:
This will repeatedly convert the Arrow Table to an array of objects, so we should do something different that’s more efficient. Either we need to move this check outside the loop (somehow, if possible), or we can use data.at in the Arrow case to avoid converting the entire table each time.